internal/spicedbx/schema.go

```diff
@@ -1,39 +1,86 @@
 package spicedbx
 
 import (
-	"strings"
+	"bytes"
+	"text/template"
+
+	"go.infratographer.com/permissions-api/internal/types"
 )
 
-func GeneratedSchema(prefix string) string {
-	schema := `
-definition PREFIX/subject {}
+var (
+	schemaTemplate = template.Must(template.New("schema").Parse(`
+{{$prefix := .Prefix}}
+definition {{$prefix}}/subject {}
+
+definition {{$prefix}}/role {
+    relation tenant: {{$prefix}}/tenant
+    relation subject: {{$prefix}}/subject
+}
 
-definition PREFIX/role {
-    relation tenant: PREFIX/tenant
-    relation subject: PREFIX/subject
+definition {{$prefix}}/tenant {
+    relation tenant: {{$prefix}}/tenant
+{{- range .ResourceTypes -}}
+{{$typeName := .Name}}
+{{range .TenantActions}}
+    relation {{$typeName}}_{{.}}_rel: {{$prefix}}/role#subject
+{{- end}}
+{{range .TenantActions}}
+    permission {{$typeName}}_{{.}} = {{$typeName}}_{{.}}_rel + tenant->{{$typeName}}_{{.}}
+{{- end}}
+{{- end}}
+}
+{{range .ResourceTypes -}}
+{{$typeName := .Name}}
+definition {{$prefix}}/{{$typeName}} {
+    relation tenant: {{$prefix}}/tenant
+{{range .TenantActions}}
+    relation {{$typeName}}_{{.}}_rel: {{$prefix}}/role#subject
+{{- end}}
+{{range .TenantActions}}
+    permission {{$typeName}}_{{.}} = {{$typeName}}_{{.}}_rel + tenant->{{$typeName}}_{{.}}
+{{- end}}
 }
+{{end}}
+`))
+)
```

Comment:
Should we already refer to the prefix as a namespace? This way it's easier to read and interpret that it's the intention.